homeassistant/components/scene/deconz.py

```diff
@@ -0,0 +1,47 @@
+"""
+Support for deCONZ scenes.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/scene.deconz/
+"""
+
+import asyncio
+import logging
+
+from homeassistant.components.deconz import DECONZ_DATA, DOMAIN
+from homeassistant.components.scene import Scene
+
+DEPENDENCIES = [DOMAIN]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up scenes for the deCONZ platform."""
+    scenes = hass.data[DECONZ_DATA].scenes
```

Comment:
Guard with `if DECONZ_DATA in hass.data:`?

Reply:
You're not supposed to get to the platform setup if the initial setup won't work. I'm removing that check on all places :)

Reply:
Oh get it, that's fine.

Reply:
@kroimon is right, people could still add it to their config.

Guard for it by checking if `discovery_info` is None:

```python
if discovery_info is None:
    return False
```

The other setup_platform will come through quickly after that.

Reply:
Im not understanding the why. I'm doing the change but would appreciate a clarification of the reasoning